lib/util/buffered_readable.js

```diff
@@ -31,7 +31,7 @@ BufferedReadable.prototype._read = function() {
   if (this._buffer.length > 0) {
     for (var i = 0; i < this._buffer.length; i++) {
       if (!this.push(this._buffer[i])) {
-        this._buffer = this._buffer.slice(i);
+        this._buffer = this._buffer.slice(i + 1);
```

Comment:
Ok it looks like in an earlier version of node, e.g. 0.10.x, the unmodified code never got triggered because it never needed to buffer with the examples I tried. In a later version of node, the same query ended up running this piece of code.